pointcloud/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSource.scala

```diff
@@ -0,0 +1,126 @@
+  sourceMetadata: Option[EPTMetadata] = None,
+  threads: Option[Int] = None,
+  targetCellType: Option[TargetCellType] = None
+) extends RasterSource {
+  @transient private[this] lazy val logger = getLogger
+
+  lazy val metadata: EPTMetadata = sourceMetadata.getOrElse(EPTMetadata(path.value))
+
+  def attributes: Map[String, String] = metadata.attributes
+  def attributesForBand(band: Int): Map[String, String] = metadata.attributesForBand(band)
+  def bandCount: Int = metadata.bandCount
+  def cellType: CellType = metadata.cellType
+  def crs: CRS = metadata.crs
+  def gridExtent: GridExtent[Long] = metadata.gridExtent
+  def name: SourceName = metadata.name
+  def resolutions: List[CellSize] = metadata.resolutions
+
+  def reprojection(targetCRS: CRS, resampleTarget: ResampleTarget, method: ResampleMethod, strategy: OverviewStrategy): IDWReprojectRasterSource =
+    IDWReprojectRasterSource(path.value, targetCRS, resampleTarget, sourceMetadata = metadata.some, threads = threads, method, targetCellType = targetCellType)
+
+  def resample(resampleTarget: ResampleTarget, method: ResampleMethod, strategy: OverviewStrategy): IDWResampleRasterSource =
+    IDWResampleRasterSource(path.value, resampleTarget, metadata.some, threads, method, targetCellType)
+
+  def read(bounds: GridBounds[Long], bands: Seq[Int]): Option[Raster[MultibandTile]] = {
+    val targetRegion = gridExtent.extentFor(bounds, clamp = false)
+    val Extent(exmin, eymin, exmax, eymax) = targetRegion.extent
+
+    val expression = ReadEpt(
+      filename   = path.value,
+      resolution = gridExtent.cellSize.resolution.some,
+      bounds     = s"([$exmin, $eymin], [$exmax, $eymax])".some,
+      threads    = threads
+    )
+
+    logger.debug(expression.asJson.spaces4)
+
+    val pipeline = expression toPipeline
+
+    try {
+      if(pipeline.validate()) {
+        pipeline.execute
+
+        val pointViews = pipeline.getPointViews().asScala.toList
+        assert(pointViews.length == 1, "Triangulation pipeline should have single resulting point view")
+
+        pointViews.headOption.map { pv =>
+          val pc = pv.getPointCloud
+          val features = Array.ofDim[PointFeature[Double]](pc.length)
+          cfor(0)(_ < pc.length, _ + 1){ i =>
+            features(i) = PointFeature[Double](Point(pc.getX(i), pc.getY(i)), pc.getZ(i))
+          }
+
+          val re = RasterExtent(
+            targetRegion,
+            bounds.width.toInt,
+            bounds.height.toInt
+          )
+
+          features
+            .toTraversable
+            .inverseDistanceWeighted(
+              re,
+              InverseDistanceWeighted.Options(
+                re.cellwidth,
+                re.cellheight,
+                0.0,  // rotation
+                3.0,  // weighting power
+                0.0,  // smoothing factor
+                re.cellSize.resolution/2, //equal weight radius
+                cellType
+              ))
```

Comment:
It looks to me this could be moved into a seaprate function / package / object. Otherwise it is just copypasted across Resample / Reproject / Common one raster sources.

Reply:
It can be done similar to what it is done in `TINRasterSource`

Reply:
This is done!